Unix/http/httpauth.c

```diff
@@ -0,0 +1,1400 @@
+                    while (!('\n' == scanp[0]
+                             && '\r' == scanp[-1])
+                           && scanp < scanlimit && !done)
+                        scanp++;
+
+                    linelimit = scanp - 1;
+
+                    linep += ORIGINAL_CONTENT_LEN;
+                    do {
+
+                        while ((isspace(*linep)
+                                || ';' == *linep)
+                               && linep < linelimit)
+                            linep++;
+                        if (Strncasecmp(linep, LENGTH_FIELD, LENGTH_FIELD_LEN) == 0) {
+                            linep += LENGTH_FIELD_LEN;
+                            original_content_length = atoi(linep);
+                            while (';' != *linep && linep < linelimit)
+                                linep++;
+                            linep++;
+                        }
+                        else if (Strncasecmp(linep, TYPE_FIELD, TYPE_FIELD_LEN) == 0) {
+                            linep += TYPE_FIELD_LEN;
+                            original_content_type = linep;
+                            while (';' != *linep && linep < linelimit)
+                                linep++;
+                            *linep++ = '\0';
+                            memcpy
+                                (original_content_type_save,
+                                 original_content_type, linep - original_content_type);
+                            original_content_type = original_content_type_save;
+                        }
+                        else if (Strncasecmp(linep, CHARSET_FIELD, CHARSET_FIELD_LEN) == 0) {
+                            linep += CHARSET_FIELD_LEN;
+                            original_encoding = linep;
+                            while (';' != *linep && linep < linelimit)
+                                linep++;
+                            *linep++ = '\0';
+                            memcpy
+                                (original_encoding_save,
+                                 original_encoding, linep - original_encoding);
+                            original_encoding = original_encoding_save;
+                        }
+
+                    } while (linep < linelimit);
+
+                    scanp = linelimit + 2;
+                }
+                else {
+                    // Bogus
+
+                    while (!('\n' == scanp[0]
+                             && '\r' == scanp[-1])
+                           && scanp < scanlimit && !done)
+                        scanp++;
+                    scanp++;
+                }
+            }
+
+        }
+        ++scanp;
+    }
+
+    if (!done) {
+        return FALSE;
+    }
+    // Alloc the new data page based on the original content size
+
+    maj_stat = gss_unwrap(&min_stat,
+                          (gss_ctx_id_t) handler->pAuthContext,
+                          &input_buffer, &output_buffer, &flags, NULL);
```

Comment:
Only one bit of the flags should be passed here, GSS_C_CONF_FLAG
https://docs.oracle.com/cd/E19683-01/816-1331/overview-137/index.html